eng/native/init-vs-env.cmd

```diff
@@ -4,10 +4,22 @@
 :: as an argument, it also initializes VC++ build environment and CMakePath.
 
 set "__VCBuildArch="
-if /i "%~1" == "x86"   (set __VCBuildArch=x86)
-if /i "%~1" == "x64"   (set __VCBuildArch=x86_amd64)
-if /i "%~1" == "arm64" (if /i "%PROCESSOR_ARCHITECTURE%" == "ARM64" (set __VCBuildArch=arm64) else (set __VCBuildArch=x86_arm64))
-if /i "%~1" == "wasm" (if /i "%PROCESSOR_ARCHITECTURE%" == "ARM64" (set __VCBuildArch=arm64) else (set __VCBuildArch=x86_amd64))
+if /i "%PROCESSOR_ARCHITECTURE%" == "AMD64" (
+    if /i "%~1" == "x64"   ( set __VCBuildArch=amd64 )
+    if /i "%~1" == "x86"   ( set __VCBuildArch=amd64_x86 )
+    if /i "%~1" == "arm64" ( set __VCBuildArch=amd64_arm64 )
+    if /i "%~1" == "wasm"  ( set __VCBuildArch=amd64 )
+) else if /i "%PROCESSOR_ARCHITECTURE%" == "ARM64" (
+    if /i "%~1" == "x64"   ( set __VCBuildArch=arm64_amd64 )
+    if /i "%~1" == "x86"   ( set __VCBuildArch=arm64_x86 )
+    if /i "%~1" == "arm64" ( set __VCBuildArch=arm64 )
+    if /i "%~1" == "wasm"  ( set __VCBuildArch=arm64 )
+) else (
+    if /i "%~1" == "x64"   ( set __VCBuildArch=x86_amd64 )
+    if /i "%~1" == "x86"   ( set __VCBuildArch=x86 )
+    if /i "%~1" == "arm64" ( set __VCBuildArch=x86_arm64 )
+    if /i "%~1" == "wasm"  ( set __VCBuildArch=x86 )
+)
```

Comment:
```suggestion
if /i "%PROCESSOR_ARCHITECTURE%" == "ARM64" (
    if /i "%~1" == "x64"   ( set __VCBuildArch=arm64_amd64 )
    if /i "%~1" == "x86"   ( set __VCBuildArch=arm64_x86 )
    if /i "%~1" == "arm64" ( set __VCBuildArch=arm64 )
    if /i "%~1" == "wasm"  ( set __VCBuildArch=arm64 )
) else (
    if /i "%~1" == "x64"   ( set __VCBuildArch=amd64 )
    if /i "%~1" == "x86"   ( set __VCBuildArch=amd64_x86 )
    if /i "%~1" == "arm64" ( set __VCBuildArch=amd64_arm64 )
    if /i "%~1" == "wasm"  ( set __VCBuildArch=amd64 )
)
```
32-bit only machines do not exist anymore. We should only care about x64 vs. arm64 build architectures (ie architecture that the build is running on). 32-bit x86 build architecture should not be a thing.

Reply:
(Same in the other script.)